### PR TITLE
Improve documentation and normalise imports in error handling

### PR DESCRIPTION
**Pull request by ushkarev**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt [hunk 1]
```diff
@@ -7,7 +7,10 @@ import org.slf4j.LoggerFactory
 import org.springframework.http.HttpStatus
 import org.springframework.http.HttpStatus.BAD_REQUEST
 import org.springframework.http.HttpStatus.CONFLICT
+import org.springframework.http.HttpStatus.FORBIDDEN
 import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import org.springframework.http.HttpStatus.NOT_FOUND
+import org.springframework.http.HttpStatus.UNAUTHORIZED
```
Comment (ushkarev) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt:13]: half of these were used as `HttpStatus.XXX` the other half imported directly as `XXX` – might as well use the latter method consistently 🤷

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt [hunk 2]
```diff
@@ -260,6 +263,6 @@ class NonAssociationAlreadyClosedException(id: Long) : Exception("Non-associatio
 
 class UserInContextMissingException : Exception("There is no user in context for this request")
 
-class OpenNonAssociationAlreadyExistsException(prisoners: List<String>) : Exception("Prisoners [$prisoners] already have open non-associations")
+class OpenNonAssociationAlreadyExistsException(prisoners: List<String>) : Exception("Prisoners $prisoners already have open non-associations")
```
Comment (ushkarev) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt:266]: `$prisoners` already formats as a list surrounded by square brackets

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt (hunk 1 above)
Comment (xoen) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt:13]: Interesting as I would have probably preferred the other way around and have a single import 😝 (e.g. use `HttpStatus.NOT_FOUND`, `HttpStatus.UNAUTHORIZED`, etc...) 

But then again, doesn't really matter.

Comment (xoen) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt:13]: (but maybe from the context is very clear these are HTTP Statuses so it's probably cleaner the way you did it 👍)

**Review by xoen — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt (hunk 1 above)
Comment (ushkarev) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt:13]: me too actually. but there were more instances in the file of the second form so i went with the more popular form 🤷

**Pull request merged**